src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -2269,5 +2269,50 @@ public sealed class SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2
         public SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2(ITestOutputHelper output) : base(output) { }
         protected override bool UseSocketsHttpHandler => true;
         protected override bool UseHttp2 => true;
+
+        [Fact]
+        public async Task Expect100Continue_NonSuccessResponse_RequestBodyNotSent()
+        {
+            string responseContent = "no no!";
+
+            await Http2LoopbackServer.CreateClientAndServerAsync(async url =>
+            {
+                using (var handler = new SocketsHttpHandler())
+                using (var invoker = new HttpMessageInvoker(handler))
+                {
+                    handler.SslOptions.RemoteCertificateValidationCallback = delegate { return true; };
+
+                    // Increase default Expect: 100-continue timeout to ensure that we don't accidentally fire the timer and send the request body.
+                    TimeSpan delay = TimeSpan.FromSeconds(5);
```

Comment:
We should never hit it, right? If you make it bigger than wait time for test (or MaxInt), it would be probably more reliable.

Reply:
Will do. I think I'll make it like 300 seconds, that should guarantee it's larger than the request timeout.